micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -198,17 +199,23 @@ public Object countedClass(ProceedingJoinPoint pjp) throws Throwable {
      * {@link Counted#recordFailuresOnly()} is set to {@code false}, a success is
      * recorded.
      * @param pjp Encapsulates some information about the intercepted area.
-     * @param counted The annotation.
      * @return Whatever the intercepted method returns.
      * @throws Throwable When the intercepted method throws one.
      */
-    @Around(value = "@annotation(counted)", argNames = "pjp,counted")
+    @Around("execution (@io.micrometer.core.annotation.Counted * *.*(..))")
     @Nullable
-    public Object interceptAndRecord(ProceedingJoinPoint pjp, Counted counted) throws Throwable {
+    public Object interceptAndRecord(ProceedingJoinPoint pjp) throws Throwable {
         if (shouldSkip.test(pjp)) {
             return pjp.proceed();
         }
 
+        Method method = ((MethodSignature) pjp.getSignature()).getMethod();
+        Counted counted = method.getAnnotation(Counted.class);
+        if (counted == null) {
+            method = pjp.getTarget().getClass().getMethod(method.getName(), method.getParameterTypes());
+            counted = method.getAnnotation(Counted.class);
+        }
+
         return perform(pjp, counted);
     }
```

Comment:
I aligned this method with `TimedAspect` not sure why were there two different approaches for the same, but the previous pointcut declaration wasn't working properly for me and this is also more consistent with `ObservedAspect` and `TimedAspect` too.